Lesson20/Les20-fercan/script.js

```diff
@@ -0,0 +1,112 @@
+// 1. Create a Person class with name, age, and gender properties, and a greet() method that logs a greeting message to the console.
+// Then create a Student class that extends the Person class and adds a major property. Override the greet() method
+// in the Student class to log a different greeting message to the console. 
+//Create 2 variables, one of type Person and one of type Student and log them to console.
+
+
+class Person {
+    constructor (name, age, gender) {
+        this._name= name;
+        this._age= age;
+        this._gender = gender;
+
+    }
+    greet () {
+    return 'Welcome to my world...My name is ${this._name}';
+}
+
+}
+const person1 = new Person('adam', 34, 'female');
+console.log('Exercise 1: ');
+console.log(person1.greet());
+
+// 2. Create a Shape class with name and color properties, and a describe() method that logs a description of the shape 
+//to the console (e.g. "This is a blue square"). Then create a Square class that extends the Shape class 
+//and adds sideLength and area properties. Override the describe() method in the Square class to log a description of the square 
+//to the console (e.g. "This is a blue square with a side length of 5 and an area of 25").
+
+ class Shape {
+    constructor (name, color) {
+        this._name = name;
+        this._color = color;
+    }
+    describe () {
+        return 'This is a ${this._color} ${this._name}';
+    }
+
+    
+ }
+
+ class Square extends Shape {
+    constructor(name, color, sideLength, area){
+        super(name, color);
+        this._sideLength = sideLength;
+        this._area = area ;
+
+    }
+    express (){
```

Comment:
You can still name this method `describe` but it can print a different message since it's a more specific class when compared to class Shape.